flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/utils/ExecNodeMetadataUtilTest.java

```diff
@@ -218,7 +331,11 @@ public void testStreamExecNodeJsonSerdeCoverage() {
         @ExecNodeMetadata(
                 name = "dummy-node",
                 version = 3,
-                consumedOptions = {"table.exec.state.ttl", "table.exec.sink.not-null-enforcer"},
+                consumedOptions = {
+                    "table.exec.state.ttl",
+                    "table.exec.sink.not-null-enforcer",
+                    "table.exec.deduplicate.mini-batch.compact-changes.enabled" // deprecated key
```

Comment:
we should not add a dependency to a real deprecated key, otherwise someone need to touch this class when removing the key, instead we should create an artificial deprecated key in this test

Reply:
Here I need real keys, because those keys, in order be returned in the persistedConfiguration they need to be in the `TableConfigOptions` and `ExecutionConfigOptions`, and I wanted to also test the deprecated keys.
A workaround would be to add some package private methods that add dummy keys to those static maps, so I can use them in the test, or do you maybe have any better suggestion?

Reply:
Please check this workaround: https://github.com/apache/flink/pull/19232/commits/0a06035b23ab6d7af82f159bc2aff52dd816364a